lib/streamlit/elements/layouts.py

```diff
@@ -87,8 +87,10 @@ def columns(
         (preferred) or just call methods directly on the returned object. See
         examples below.
 
+        Columns can only be placed inside other columns up to one level of nesting.
+
         .. warning::
-            Currently, you may not put columns inside another column.
+            Columns cannot be placed inside other columns in the sidebar. This is only possible in the main area of the app.
```

Comment:
@sfc-gh-tszerszen I'm not a 100% sure we want to call the non-sidebar area of the app as the `main area`.  I've asked the design team if there's a technical term for it in this [Slack thread](https://snowflake.slack.com/archives/C03ASKRP28G/p1673509897021869). I'll update this PR and approve once we get a response

Reply:
Ok the consensus is to call it the "main area", just like you've done in this PR 😄